src/core/ext/transport/chaotic_good/client/chaotic_good_connector.cc

```diff
@@ -0,0 +1,329 @@
+                             frame.headers = std::move(metadata);
+                             auto write_buffer =
+                                 frame.Serialize(self->hpack_compressor_.get());
+                             return self->data_endpoint_->Write(
+                                 std::move(write_buffer));
+                           },
+                           []() -> absl::Status { return absl::OkStatus(); });
+                       auto read_promise = TrySeq(
+                           self->data_endpoint_->ReadSlice(
+                               FrameHeader::frame_header_size_),
+                           [self = self](Slice slice) mutable {
+                             GPR_ASSERT(self->data_endpoint_ != nullptr);
+                             auto frame_header_ = std::make_shared<FrameHeader>(
+                                 FrameHeader::Parse(
+                                     reinterpret_cast<const uint8_t*>(
+                                         GRPC_SLICE_START_PTR(slice.c_slice())))
+                                     .value());
+                             auto frame_header_length =
+                                 frame_header_->GetFrameLength();
+                             return self->data_endpoint_->Read(
+                                 frame_header_length);
+                           },
+                           []() { return absl::OkStatus(); });
+                       return TrySeq(
+                           TryJoin(std::move(write_promise),
+                                   std::move(read_promise)),
+                           []() -> absl::Status { return absl::OkStatus(); });
+                     }),
+                 TrySeq(Sleep(Timestamp::Now() + Duration::Seconds(5)),
+                        []() -> absl::Status {
+                          return absl::DeadlineExceededError(
+                              "Data endpoint connect deadline excced.");
+                        })),
+            []() { return absl::OkStatus(); });
+      });
+  auto send_setting_frames = TrySeq(
+      [self = self]() mutable {
+        // Serialize setting frame.
+        SettingsFrame frame;
+        // frame.header set connectiion_type: control
+        ClientMetadataHandle metadata =
+            self->arena_->MakePooled<ClientMetadata>(self->arena_.get());
+        metadata->Set(ChaoticGoodConnectionTypeMetadata(),
+                      Slice::FromCopiedString("control"));
+        auto connection_type =
+            metadata->get_pointer(ChaoticGoodConnectionTypeMetadata())->Ref();
+        frame.headers = std::move(metadata);
+        auto write_buffer = frame.Serialize(self->hpack_compressor_.get());
+        ;
+        return self->control_endpoint_->Write(std::move(write_buffer));
+      },
+      []() { return absl::OkStatus(); });
+
+  auto activity = MakeActivity(
+      TrySeq(TryJoin(std::move(read_setting_frames),
```

Comment:
`TryJoin` here now needs to specify the output status type... so in this case you need to write `TryJoin<absl::StatusOr>(...)`

Reply:
Done.